src/getitem.cpp

```diff
@@ -0,0 +1,106 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <miopen/getitem.hpp>
+#include <miopen/kernel_cache.hpp>
+#include <miopen/float_equal.hpp>
+#include <miopen/check_numerics.hpp>
+#include <miopen/tensor.hpp>
+#include <miopen/datatype.hpp>
+#include <miopen/item/invoke_params.hpp>
+#include <miopen/item/solvers.hpp>
+#include <miopen/find_solution.hpp>
+
+namespace miopen {
+
+std::size_t GetGetitemWorkspaceSize(Handle& handle,
+                                    int32_t indexCount,
+                                    const TensorDescriptor* const* indexDescs)
+{
+    auto ctx           = ExecutionContext{&handle};
+    const auto problem = item::ProblemDescription{indexCount, indexDescs};
+
+    const auto algo    = AlgorithmName{"GetitemBackward"};
+    const auto solvers = solver::SolverContainer<solver::item::GetitemBackward>{};
+
+    auto pair_size_vector = solvers.GetWorkspaceSizes(ctx, problem);
```

Comment:
[Q] Do you know that `SolverContainer::GetWorkspaceSizes()` is inadequate for primitives that use `SolverContainer::ExecutePrimitive()`?

Reply:
This only works as long as there is only one solver in the primitive, or all the GetWorkspaceSize methods from the solvers return the same value.

Reply:
I had a wrong understanding of the structure of the solver. fixed it.

Reply:
[Notice] @seungmanhan The idea is to make `SolverContainer::GetWorkspaceSize()` (notice the singular "size" in the name) that should return WS size for the first applicable solver, and to use it instead of `SolverContainer::GetWorkspaceSizes()`.

However, current code is Ok so far, provided that we have only one solver. When time permits I'll create `SolverContainer::GetWorkspaceSize()` and let you know.

Reply:
@seungmanhan I highly recommend:
- getting back the get-workspace-size-related changes
-  applying the patch that makes existing `SolverContainer::GetWorkspaceSizes()` suitable for the simple primitives ([pr2883-getworkspacesizes-for-simple-primitives.diff.txt](https://github.com/ROCm/MIOpen/files/15462175/pr2883-getworkspacesizes-for-simple-primitives.diff.txt), also it contains the example patch for GWSS in **_sum_**), and
- changing the call of `SolverContainer::GetWorkspaceSizes()` to:
```C++
const auto wss = solvers.GetWorkspaceSizes(ctx, problem, true);
return wss.empty() ? static_cast<size_t>(0) : wss.front().second;
```
Why: `true` tells it to NOT skip solvers that do not require workspace (but return a pair `{solvername, 0}` instead). This is necessary for cases when the 1st applicable primitive does not require workspace.

This is important to keep implementations of simple primitives in sync with the Solver/Solution approach used in the library.

Reply:
[resolved]